src/helpers.rs

```diff
@@ -75,12 +74,15 @@ const UNIX_IO_ERROR_TABLE: &[(&str, std::io::ErrorKind)] = {
         ("ETIMEDOUT", TimedOut),
         ("ETXTBSY", ExecutableFileBusy),
         ("EXDEV", CrossesDevices),
-        // The following have two valid options. We have both for the forwards mapping; only the
-        // first one will be used for the backwards mapping.
+        // The following have multiple valid options. We have all of them for the forwards mapping;
+        // only the first one will be used for the backwards mapping.
         ("EPERM", PermissionDenied),
         ("EACCES", PermissionDenied),
         ("EWOULDBLOCK", WouldBlock),
         ("EAGAIN", WouldBlock),
+        ("ENOTSUP", Unsupported),
+        ("EOPNOTSUPP", Unsupported),
```

Comment:
Further up the comment says this mapping should be in sync with the one in the standard library. The new ones you are adding here are not present in the standard library. Why should we add them here?

Reply:
> Why should we add them here?

I would prefer that we do _something_ in this situation. The motivating example is the `mmap` call I pasted above, which reports an error string with wording that suggests there is a bug in Miri.

Reply:
Maybe Miri's mmap shouldn't set `ENOTSUP` if that error code is not supported (heh) by Miri itself.

Reply:
Though usually what we do in that case is just `throw_unsup_format!`, which can give a lot more context and so be much more informative. Is there a reason mmap doesn't do that?